docs/emitters/code-generation/server/server-code-generation.md

```diff
@@ -0,0 +1,375 @@
+                    }
+
+                    return Task.FromResult<IActionResult>(Ok(comment));
+                }
+            }
+        }
+        ```
+
+        **Important**: if your controller file is not automatically generated as `CommentOpsOperationsController.cs`, you will have to manually add in `[ApiController]` and inherit the base class `CommentOpsOperationsControllerBase`.
+
+*   **Node.js/express example:**
+
+    [Placeholder: Detailed instructions on setting up a Node.js/Express project, integrating the generated code, and running the server will be added here.]
+
+    1.  **Create a basic express app:** Use the Express generator to create a new project:
+
+        ```bash
+        npm install -g express-generator
+        express my-express-app
+        cd my-express-app
+        npm install
+        ```
+
+    2.  **Install additional dependencies:** You may need additional dependencies depending on your API requirements (e.g., `body-parser` for parsing request bodies).
+
+        ```bash
+        npm install body-parser --save
+        ```
+
+    3.  **Integrate generated code:** Copy the contents of the `servers/node/tsp-output/@typespec/http-server-javascript` directory into your Express project. You will likely need to adapt the generated code to fit the Express routing and middleware structure.
+
+    4.  **Configure routes:** Define your API routes in your Express app, mapping them to the generated controller functions.
+
+    5.  **Run the server:** Start the Express server:
+
+        ```bash
+        npm start
+        ```
+
+## Cli: running the project and adding necessary code:
+
+Now that you have the base controller, models and service interface generated, the next step is to implement the code with a service scaffolding and business logic for a runnable server. Please follow the instruction below:
+
+1.  **ASP.NET webapi server (C#):**
+
+    1.  Navigate to the server directory:
+
+        ```bash
+        cd servers/aspnet
+        ```
+
+    2.  Run the server:
+
+        ```bash
+        dotnet run
+        ```
+
+    3.  Open your browser and navigate to `http://localhost:[PORT#]/swagger/index.html` (replace `[PORT#]` with the actual port number shown in the console output). You should see the Swagger UI, allowing you to test your API endpoints.
+
+## Using the vscode extension
+
+The TypeSpec Extension for VSCode offers a convenient way to generate server stubs directly from your IDE.
+
+1.  **Install the extension:** Search for "TypeSpec" in the VSCode Extensions Marketplace and install the official TypeSpec extension.
+
+2.  **Create a TypeSpec project:**
+    *   Click “Create TypeSpec Project” in the EXPLORE sidebar.
+        *   `[Placeholder: Screenshot showing "Create TypeSpec Project" in the VSCode Explorer sidebar]`
+    *   Select a project root folder.
+    *   Provide the required inputs via Quick Picks, similar to using `tsp init` on the command line.
+
```

Comment:
May use this gif - ![vscode_project_scaffolding.gif](https://raw.githubusercontent.com/microsoft/typespec/main/website/src/content/docs/docs/images/vscode_project_scaffolding.gif)

Reply:
👍